src/serialize.h

```diff
@@ -139,27 +140,27 @@ template<typename Stream> inline uint64_t ser_readdata64(Stream &s)
 }
 inline uint64_t ser_double_to_uint64(double x)
 {
-    union { double x; uint64_t y; } tmp;
-    tmp.x = x;
-    return tmp.y;
+    uint64_t tmp;
+    std::memcpy(&tmp, &x, sizeof(x));
```

Comment:
Maybe add `static_assert(sizeof(tmp) == sizeof(x), "double and uint64_t should be the same size");`?
(this might be redundant with other places but I think it's also helpful as documentation)

(Same for the rest of the functions)

Reply:
I swear I wanted to do that, but then I was afraid people would find it paranoid 🤐 

You sure I should go for that?

Reply:
Explicitly stating assumptions is good and `static_assert`:s cannot hurt :)  Go for it! :)

FWIW:

https://github.com/bitcoin/bitcoin/blob/68e841e0af223a220d1f037e4c5680c1b228aa3e/src/compat/assumptions.h#L43-L46

Reply:
Done.